datasets/air_dialogue/air_dialogue.py

```diff
@@ -0,0 +1,316 @@
+                                if "field_value" not in data["search_info"]
+                                else data["search_info"]["field_value"],
+                                "timestmamp": -1
+                                if "timestmamp" not in data["search_info"]
+                                else data["search_info"]["timestmamp"],
+                            },
+                        ]
+                    )
+
+                    yield id_, {
+                        "action": {key: data["action"][key] for key in data["action"]},
+                        "intent": intent,
+                        "timestamps": data["timestamps"],
+                        "dialogue": data["dialogue"],
+                        "expected_action": {key: data["expected_action"][key] for key in data["expected_action"]},
+                        "search_info": search_info,
+                        "correct_sample": data["correct_sample"],
+                    }
+
+                else:
+
+                    kb = [
+                        {
+                            "airline": "" if "airline" not in data["kb"] else data["kb"]["airline"],
+                            "class": "" if "class" not in data["kb"] else data["kb"]["class"],
+                            "departure_airport": ""
+                            if "departure_airport" not in data["kb"]
+                            else data["kb"]["departure_airport"],
+                            "departure_day": "" if "departure_day" not in data["kb"] else data["kb"]["departure_day"],
+                            "departure_month": ""
+                            if "departure_month" not in data["kb"]
+                            else data["kb"]["departure_month"],
+                            "departure_time_num": -1
+                            if "departure_time_num" not in data["kb"]
+                            else data["kb"]["departure_time_num"],
+                            "flight_number": -1 if "flight_number" not in data["kb"] else data["kb"]["flight_number"],
+                            "num_connections": -1
+                            if "num_connections" not in data["kb"]
+                            else data["kb"]["num_connections"],
+                            "price": -1 if "price" not in data["kb"] else data["kb"]["price"],
+                            "return_airport": ""
+                            if "return_airport" not in data["kb"]
+                            else data["kb"]["return_airport"],
+                            "return_day": "" if "return_day" not in data["kb"] else data["kb"]["return_day"],
+                            "return_month": "" if "return_month" not in data["kb"] else data["kb"]["return_month"],
+                            "return_time_num": -1
+                            if "return_time_num" not in data["kb"]
+                            else data["kb"]["return_time_num"],
+                        }
+                    ]
```

Comment:
It looks like there's only one element in this list.

Is this expected ?

In the examples in the dataset card I can see several flights in the `kb` field.

Reply:
`kb` is a list of available flights between the `departure airport` (_IAD_ )and the `return airport` (_DTW_). Each flight information is available as a dict. 
am I missing something?

Reply:
I see. For example in the example provided in the dataset card there are many available flights:
```
{"kb": [
    {"return_airport": "DTW", "airline": "Spirit", "departure_day": "12", "departure_airport": "IAD", "flight_number": 1000, "departure_month": "June", "departure_time_num": 17, "class": "economy", "return_time_num": 2, "return_month": "June", "return_day": "14", "num_connections": 1, "price": 200},
    {"return_airport": "DTW", "airline": "Frontier", "departure_day": "12", "departure_airport": "IAD", "flight_number": 1001, "departure_month": "June", "departure_time_num": 0, "class": "business", "return_time_num": 15, "return_month": "June", "return_day": "13", "num_connections": 0, "price": 500},
    {"return_airport": "DTW", "airline": "JetBlue", "departure_day": "12", "departure_airport": "IAD", "flight_number": 1002, "departure_month": "June", "departure_time_num": 0, "class": "business", "return_time_num": 13, "return_month": "June", "return_day": "13", "num_connections": 1, "price": 600},
    ... 
    ]
"reservation": 0}
```

However in line 283 you are defining the `kb` variable as a list of exactly one element right ?

Reply:
I believe I am missing some thing here, how do I define it to accept multiple values? 

also just realised the data_loader its not loading the data correctly. Its storing default values instead of the actual data!

Reply:
Ok I see the problem ! That's because data["kb"] is a list, not a dictionary.

Therefore you can parse it to get all the available flights this way for example:

```python
kb = [
    {
        "airline": flight["airline"],
        "class": flight["class"],
        "departure_airport": flight["departure_airport"],
        "departure_day": flight["departure_day"],
        "departure_month": flight["departure_month"],
        "departure_time_num": flight["departure_time_num"],
        "flight_number": flight["flight_number"],
        "num_connections": flight["num_connections"],
        "price": flight["price"],
        "return_airport": flight["return_airport"],
        "return_day": flight["return_day"],
        "return_month": flight["return_month"],
        "return_time_num": flight["return_time_num"],
    }
    for flight in data["kb"]
]
```

If you find out that there are missing values for some fields, you can use default values.

Reply:
got it. Updated.

Reply:
thanks !